fastlane_core/lib/fastlane_core/tool_collector.rb

```diff
@@ -2,16 +2,49 @@ module FastlaneCore
   class ToolCollector
     HOST_URL = "https://fastlane-enhancer.herokuapp.com"
 
+    # This is the original error reporting mechanism, which has always represented
+    # either controlled (UI.user_error!), or uncontrolled (UI.crash!, anything else)
+    # exceptions.
+    #
+    # Thus, if you call `did_crash`, it will record the failure both here, and in the
+    # newer, more specific `crash` field.
     attr_reader :error
 
+    # This is the newer field for tracking only uncontrolled exceptions.
+    #
+    # This is written to only when `did_crash` is called, and therefore excludes
+    # controlled exceptions.
+    attr_reader :crash
+
     def did_launch_action(name)
       name = name.to_sym
-      launches[name] += 1 if is_official?(name)
+
+      if is_official?(name)
+        launches[name] += 1
+        versions[name] ||= determine_version(name)
+      end
     end
 
+    # Call when the problem is a caught/controlled exception (e.g. via UI.user_error!)
     def did_raise_error(name)
```

Comment:
Maybe we should rename this method to `did_raise_user_error` and provide an `:alias` for the old method to be backwards compatible?